React/creating-components.md

```diff
@@ -0,0 +1,79 @@
+# Creating Components
+
+## Learning Goals
+- Create our own component
+- Link and render a component from an existing component
+
+Now that we've gotten started with a basic React application, we want to be able to extend it with our own components. We create new components within our applications to encapsulate logic and HTML related to a specific concept or portion of our site. Note that up until this point, we have had very clear lines between HTML, CSS and JavaScript because they're usually in separate files. With React components, these lines can get a little bit blurry since we will potentially have all of these things in a single component file.
+
+Each component will be a small, reusable piece of code that manages their own **content, presentation and behavior**. 
+
+### Overview
+It is important to understand what is happening currently within the React application so we can model the same pattern with a custom component. Right now, our React application renders the `App` component from the `index.js` file and puts it in the DOM element where the ID is "root". Normally, this is only done **once** in a React application, so we are going to leave this code alone as we build on.
+
+Next, our `App` component contains a render function will which render a `header` and a `p` using JSX.
+
+The powerful thing about React is that **any component can render any other component**. That is, we can (and should) **nest our components** within one another to create the most effective React application design.
+
+### Context
+As we continue using this application, we are going to be creating a React application that manages our list of Ada students! We are going to use different components to handle different portions of our student listing.
```

Comment:
It might be helpful to the global learners to include a link here to a completed version of the app, so that we know what we're working toward.